lib/GcInfo/GcInfo.cpp

```diff
@@ -0,0 +1,271 @@
+//===-------- include/gcinfo/gcinfo.cpp -------------------------*- C++ -*-===//
+//
+// LLILC
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// \brief Implements the generation of CLR GCTables in LLILC
+///
+//===----------------------------------------------------------------------===//
+
+#include "earlyincludes.h"
+#include "GcInfo.h"
+#include "gcinfoencoder.h"
+#include "jitpch.h"
+#include "LLILCJit.h"
+#include "Target.h"
+#include "llvm/Object/StackMapParser.h"
+#include <map>
+#include <vector>
+#include <sstream>
+
+using namespace llvm;
+using namespace std;
+
+GCInfo::GCInfo(LLILCJitContext *JitContext, uint8_t *LLVMStackMapData,
+               uint8_t *CodeBlockStart) {
+  this->JitContext = JitContext;
+  this->LLVMStackMapData = LLVMStackMapData;
+  this->CodeBlockStart = CodeBlockStart;
+
+  this->Encoder = new GcInfoEncoder(JitContext->JitInfo, JitContext->MethodInfo,
+                                    new GcInfoAllocator());
```

Comment:
Instead of allocating a `GcInfoAllocator` on the heap, I'd simply add a field of type `GcInfoAllocator` to `GCInfo` and pass in a pointer to that field.